src/interface/src/app/plan/create-scenarios/constraints-panel/constraints-panel.component.ts

```diff
@@ -174,13 +233,50 @@ export class ConstraintsPanelComponent {
     }
   }
 
-  private constraintsFormValidator(
+  /**
+   * checks that one of budget or treatment area constraints is provided.
+   * @param constraintsForm
+   * @private
+   */
+  private budgetOrAreaRequiredValidator(
     constraintsForm: AbstractControl
   ): ValidationErrors | null {
-    // Only one of budget or treatment area constraints is required.
     const maxCost = constraintsForm.get('budgetForm.maxCost');
     const maxArea = constraintsForm.get('physicalConstraintForm.maxArea');
     const valid = !!maxCost?.value || !!maxArea?.value;
-    return valid ? null : { budgetOrAreaRequired: true };
+    return valid ? null : { [customErrors.budgetOrAreaRequired]: true };
+  }
+
+  /**
+   * Checks that the maxBudget is enough for the selected estimatedCost per acre
+   * @param planningAreaAcres
+   * @private
+   */
+  private totalBudgetedValidator(planningAreaAcres: number): ValidatorFn {
+    return (constraintsForm: AbstractControl): ValidationErrors | null => {
+      const maxCost = constraintsForm.get('budgetForm.maxCost')?.value;
+      const estCostPerAcre = constraintsForm.get('budgetForm.estimatedCost')
+        ?.value;
+      if (!!maxCost) {
+        const totalBudgetedAcres = maxCost / estCostPerAcre;
+        return totalBudgetedAcres < planningAreaAcres * 0.2
```

Comment:
nit: I know all of these min/max calculations are basic arithmetic, but I wonder if we shouldn't still move them to somewhere like `plan-helpers.ts` instead of the component, for clarity, testing, or maintenance -- especially if we decide to change this from 20% to 25% or something similar.

Reply:
great point, will do